models/optional_modules/core_web_vitals/snowflake/snowplow_web_vital_measurements.sql

```diff
@@ -0,0 +1,262 @@
+{{
+  config(
+    enabled=var("snowplow__enable_cwv", false) and target.type == 'snowflake' | as_bool(),
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+with by_url_and_device as (
+
+  select distinct
+
+    page_url,
+    device_class,
+    'all' as country,
+    'last {{var('snowplow__cwv_days_to_measure')|string }} days' as time_period,
+    count(*) over (partition by page_url, device_class) as page_view_count,
+    nullif(percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by lcp) over (partition by page_url, device_class), 0) as lcp_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by fid) over (partition by page_url, device_class) as fid_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by cls) over (partition by page_url, device_class) as cls_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by ttfb) over (partition by page_url, device_class) as ttfb_{{ var('snowplow__cwv_percentile') }}p,
+    'by_url_and_device' as measurement_type
+
+  from {{ ref('snowplow_web_vitals') }}
+
+  where derived_tstamp >= {{ snowplow_utils.timestamp_add('day', '-'+var('snowplow__cwv_days_to_measure')|string, snowplow_utils.current_timestamp_in_utc()) }}
+
+)
+
+, overall as (
+
+  select distinct
+
+    'all' as page_url,
+    'all' as device_class,
+    'all' as country,
+    'last {{var('snowplow__cwv_days_to_measure')|string }} days' as time_period,
+    count(*) as page_view_count,
+    nullif(percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by lcp), 0) as lcp_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by fid) as fid_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by cls) as cls_{{ var('snowplow__cwv_percentile') }}p,
+    percentile_cont(0.{{ var('snowplow__cwv_percentile') }}) within group (order by ttfb) as ttfb_{{ var('snowplow__cwv_percentile') }}p,
+    'overall' as measurement_type
+
+  from {{ ref('snowplow_web_vitals') }}
+
+  where derived_tstamp >= {{ snowplow_utils.timestamp_add('day', '-'+var('snowplow__cwv_days_to_measure')|string, snowplow_utils.current_timestamp_in_utc()) }}
+
```

Comment:
Same question around distinct, although I think here you get the same result without a distinct you aren't actually grouping by any columns...